language-plutus-core/src/Language/PlutusCore/TypeSynthesis.hs

```diff
@@ -251,7 +231,8 @@ typeOf (LamAbs x _ dom body)                     = do
 -- [infer| error ty : vTy]
 typeOf (Error x ty)                              = do
     kindCheckM x ty $ Type ()
-    normalizeType $ void ty
+    (TypeConfig _ _ gas) <- ask
+    normalizeType gas $ void ty
```

Comment:
This looks like a mistake - I think this should be `normalizeTypeOpt`.

Reply:
This definitely looks wrong. I remember I was carefully handling this case, not sure how it ended like that.